### PR TITLE
MILESTONE 4: Planetary Sectors

### PR DESCRIPTION
**Pull request by willkotheimer**
THIS PR FINISHES MILESTONE 4: Sectors

## Description

Planetary Sectors: 🪐 

Our team created a CRUD app with firebase to be able to maintain the area of the website that deals with planetary sectors.

## Related Issue

https://github.com/nss-evening-cohort-13/star-destroyer-dashboard-star-destroyer-dashboard/issues/38

## Motivation and Context

The purpose of this is to use axios and firebase to create a CRUD

## How Can This Be Tested?

git fetch oirigin development
git checkout development
npm start

## URL:

[https://star-destroyer-a0766.web.app](https://star-destroyer-a0766.web.app)

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### CODE REVIEW

File: src/javascripts/components/cards/sectorCards.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+  let buttons = '';
+  const user = firebase.auth().currentUser;
+  if (user) {
+    buttons += `<a href='#' id='${object.firebaseKey}' class='btn btn-primary edit-sectors'>Edit</a>
+      <a id='delete__${object.firebaseKey}' href='#' class='delete btn btn-danger'>Delete</a>
+  `;
+  }
+
+  const domString = $('#app').append(`
+    <div class="weapons" id="sectors-${object.firebaseKey}">
```
Comment (brownabs) [inline src/javascripts/components/cards/sectorCards.js]: Why are you using class weapons for the sectors?

File: src/javascripts/components/forms/sectorsForm.js [hunk 2]
```diff
@@ -0,0 +1,67 @@
+import sectorData from '../../helpers/data/sectorData';
+import systemsData from '../../helpers/data/systemsData';
+
+const getNewSector = () => ({
+  name: $('#sector-name').val() || false,
+  system_id: $('#system_id').val() || false
```
Comment (brownabs) [inline src/javascripts/components/forms/sectorsForm.js:6]: Are you wanting to display the system associated with the sector? I'm confused as to why it is asked, but not displayed.

File: src/javascripts/components/forms/updateSector.js [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+      <h2>Update sector</h2>
+      <div id="success-message"></div>
+      <form>
+        <div id="error-message"></div>
+        <div class="form-group">
+          <label for="name">Name</label>
+          <input type="text" class="form-control" id="name" value="${sectorObject.name}">
+        </div>
+        <div class="form-group">
+          <label for="systems">System</label>
```
Comment (brownabs) [inline src/javascripts/components/forms/updateSector.js:15]: Again, why would you ask for system if you wouldn't display it somewhere on the sector card. Just curious!

**Review by brownabs — CHANGES_REQUESTED** [review]

File: src/javascripts/components/forms/updateSector.js (hunk 3 above)
Comment (willkotheimer) [reply src/javascripts/components/forms/updateSector.js:15]: This is part of our stretch goal. Just laying the groundwork here. We can take it out if needed.

File: src/javascripts/components/forms/sectorsForm.js (hunk 2 above)
Comment (willkotheimer) [reply src/javascripts/components/forms/sectorsForm.js:6]: System ids are part of the strech goal. But we can take it out if needed.

File: src/javascripts/components/cards/sectorCards.js (hunk 1 above)
Comment (willkotheimer) [reply src/javascripts/components/cards/sectorCards.js]: That's copypasta. You are right it should be removed.

File: src/javascripts/components/forms/sectorsForm.js (hunk 2 above)
Comment (willkotheimer) [reply src/javascripts/components/forms/sectorsForm.js:6]: Let us know what you would like to see.

File: src/javascripts/components/forms/updateSector.js (hunk 3 above)
Comment (brownabs) [reply src/javascripts/components/forms/updateSector.js:15]: No need if there is a plan, go for it!

**Comment by brownabs** [comment]
Ok, change your class and then it will be ready for approval

**Comment by willkotheimer** [comment]
Thanks! I'll get it real quick!

**Comment by willkotheimer** [comment]
It's merged in! Thanks for approving it!

File: src/javascripts/components/forms/sectorsForm.js (hunk 2 above)
Comment (drteresavasquez) [reply src/javascripts/components/forms/sectorsForm.js:6]: I agree with @brownabs stretch goals are for future development. Take it out.

**Review by wppattonjr — APPROVED** [review]
Closing as this has been approved.

**Pull request closed without merging**